### PR TITLE
enable more linters

### PR DESCRIPTION
**Pull request by faddat**
Closes: #1024 

## Description

This PR:

* Enables many new linters in golangci-lint
* Uses some automated code cleanup features provided by goland's inspect feature
* Aims not to change functionality in any way

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: cmd/osmosisd/cmd/balances_from_state_export.go [hunk 1]
```diff
@@ -177,18 +177,22 @@ Example:
 			}
 
 			authGenesis := authtypes.GenesisState{}
-			clientCtx.JSONCodec.MustUnmarshalJSON(genState["auth"], &authGenesis)
-			accounts, err := authtypes.UnpackAccounts(authGenesis.Accounts)
-			if err != nil {
-				panic(err)
-			}
-			accounts = authtypes.SanitizeGenesisAccounts(accounts)
+			clientCtx.Codec.MustUnmarshalJSON(genState["auth"], &authGenesis)
```
Comment (faddat) [inline cmd/osmosisd/cmd/balances_from_state_export.go:180]: @ValarDragon this is questionable-- think it is right?  Basically, didn't see any difference with it enabled.

File: x/lockup/keeper/store.go [hunk 2]
```diff
@@ -50,19 +50,6 @@ func syntheticLockTimeStoreKey(lockID uint64, synthDenom string, endTime time.Ti
 		[]byte(synthDenom))
 }
 
-// getLockRefs get lock IDs specified on the prefix and timestamp key
```
Comment (faddat) [inline x/lockup/keeper/store.go:53]: removing this function seemed to change nothing

File: cmd/osmosisd/cmd/balances_from_state_export.go [hunk 3]
```diff
@@ -177,18 +177,22 @@ Example:
 			}
 
 			authGenesis := authtypes.GenesisState{}
-			clientCtx.JSONCodec.MustUnmarshalJSON(genState["auth"], &authGenesis)
```
Comment (faddat) [inline cmd/osmosisd/cmd/balances_from_state_export.go:180]: I think that this was used at initial genesis, and is no longer needed.

File: cmd/osmosisd/cmd/testnetify/state_export_to_testnet.go [hunk 4]
```diff
@@ -7,7 +7,6 @@ import (
 
 var (
 	flagTestnetParams = "empty"
-	valConsBech32     = "osmovalcons"
```
Comment (faddat) [inline cmd/osmosisd/cmd/testnetify/state_export_to_testnet.go:10]: unused

File: cmd/osmosisd/cmd/testnetify/state_export_to_testnet.go [hunk 5]
```diff
@@ -44,8 +43,6 @@ func defaultTestnetParams() TestnetParams {
 	consensusPubkey := "2OpBuqaXvXQ+lSxAoT1S7Jfyr56KiakTzvuFiuJK+X4="
 	// mnemonic kitchen comic flower drip sick prize account cheese truth income weekend nominee segment punch call satisfy captain earth ethics wasp clump tunnel orchard advance
 	operatorPubkey := "osmopub1addwnpepq0wv95fxk32z90u42t3df2l5pdtngvg20rkalv5vt2d7n5q5ekk35d8hh20"
-	operatorValOper := "osmovaloper1qye772qje88p7ggtzrvl9nxvty6dkuusvpqhac"
```
Comment (faddat) [inline cmd/osmosisd/cmd/testnetify/state_export_to_testnet.go:47]: unused

File: x/claim/client/rest/rest.go [hunk 6]
```diff
@@ -14,8 +14,3 @@ const (
 func RegisterRoutes(clientCtx client.Context, r *mux.Router) {
 }
 
-func registerQueryRoutes(clientCtx client.Context, r *mux.Router) {
```
Comment (faddat) [inline x/claim/client/rest/rest.go:17]: I took this out across many files.

File: x/gamm/simulation/operations.go [hunk 7]
```diff
@@ -70,19 +70,6 @@ func WeightedOperations(
 	}
 }
 
-func genFuturePoolGovernor(r *rand.Rand, addr sdk.Address, tokenList []string) string {
```
Comment (faddat) [inline x/gamm/simulation/operations.go:73]: unused

File: .golangci.yml [hunk 8]
```diff
@@ -3,14 +3,6 @@ run:
...
 #   timeout: 5m
 
-linters:
-  disable-all: true
-  enable:
-    - errcheck
-    - gofmt
-    # - goimports
-    # - golint
-    - maligned
```
Comment (ValarDragon) [inline .golangci.yml:13]: I'd significantly prefer the approach of disabling all, and manually re-enabling, but doing so to explicitly add what we want to add

Comment (faddat) [reply .golangci.yml:13]: ah, that way there's a list and we know which ones!

ok, can do :)

Comment (faddat) [reply .golangci.yml:13]: it is kinda hilarious, the degree to which you made a truly excellent call here.

File: .golangci.yml [hunk 9]
```diff
@@ -4,13 +4,90 @@ run:
 #   timeout: 5m
 
 linters:
+  # Disable all linters.
+  # Default: false
   disable-all: true
+  # Enable specific linter
+  # https://golangci-lint.run/usage/linters/#enabled-by-default-linters
   enable:
+    - asciicheck
```
Comment (ValarDragon) [inline .golangci.yml:13]: oh wow, I was considering adding a grep command to CI for this. Glad it exists in golangci!

File: .golangci.yml [hunk 10]
```diff
@@ -4,13 +4,90 @@ run:
...
+    - forbidigo
+    - forcetypeassert
+    - funlen
+    - gci
+    - gochecknoglobals
+    - gochecknoinits
+    - gocognit
+    - goconst
+    - gocritic
+    - gocyclo
```
Comment (ValarDragon) [inline .golangci.yml:38]: Lets remove `gocyclo`, we had a terrible experience with it back in the SDK

File: .golangci.yml [hunk 11]
```diff
@@ -4,13 +4,90 @@ run:
...
+    - lll
+    - makezero
+    - misspell
+    - nakedret
+    - nestif
+    - nilerr
+    - nilnil
+#    - nlreturn <- disabled as it doesn't auto-fix something simple and doesn't add anything useful
+    - noctx
+    - nolintlint
```
Comment (ValarDragon) [inline .golangci.yml:65]: Lets remove this one, nolints are quite useful!

We can always just grep for em if we need, but this in CI actually makes development harder

File: .golangci.yml [hunk 12]
```diff
@@ -4,13 +4,90 @@ run:
...
+    - testpackage
+    - thelper
+    - tparallel
+    - typecheck
+    - unconvert
+    - unparam
+    - unused
+    - varcheck
+#    - varnamelen
+    - wastedassign
```
Comment (ValarDragon) [inline .golangci.yml:87]: Lets remove this one, you often have to use this pattern for fighting the compiler

`_ = a`, when you want `a` to be named despite being unused for readability purposes

**Comment by ValarDragon** [comment]
Thanks so much for working on this btw!

This is the sort of thing that becomes really helpful as the codebase grows!

File: .golangci.yml [hunk 13]
```diff
@@ -4,13 +4,90 @@ run:
...
+    - makezero
+    - misspell
+    - nakedret
+    - nestif
+    - nilerr
+    - nilnil
+#    - nlreturn <- disabled as it doesn't auto-fix something simple and doesn't add anything useful
+    - noctx
+#    - nolintlint <- disabled because we use nolint in some places
+    - paralleltest
```
Comment (ValarDragon) [inline .golangci.yml:66]: https://github.com/kunwardeep/paralleltest

I don't think I understand what this one does o_O

File: .golangci.yml [hunk 14]
```diff
@@ -4,13 +4,90 @@ run:
...
+  # https://golangci-lint.run/usage/linters/#enabled-by-default-linters
   enable:
+    - asciicheck
+    - bidichk
+    - bodyclose
+    - contextcheck
+    - cyclop
+    - deadcode
+    - depguard
+    - dogsled
```
Comment (ValarDragon) [inline .golangci.yml:20]: This one seems a bit odd to me

File: .golangci.yml [hunk 15]
```diff
@@ -4,13 +4,90 @@ run:
...
+  # Default: false
   disable-all: true
+  # Enable specific linter
+  # https://golangci-lint.run/usage/linters/#enabled-by-default-linters
   enable:
+    - asciicheck
+    - bidichk
+    - bodyclose
+    - contextcheck
+    - cyclop
```
Comment (ValarDragon) [inline .golangci.yml:17]: Ah this should probably be disabled for the same reason as gocyclo

Comment (faddat) [reply .golangci.yml:17]: done :)

File: .golangci.yml (hunk 14 above)
Comment (faddat) [reply .golangci.yml:20]: doesn't it?

ps we are now talking via vscode.  it is neat.

File: .golangci.yml (hunk 13 above)
Comment (faddat) [reply .golangci.yml:66]: maybe encourages you to write tests that are parallelized?

Comment (faddat) [reply .golangci.yml:66]: if so I wasn't really aware of the feature before -- cut it?

Comment (ValarDragon) [reply .golangci.yml:66]: yeah SGTM

**Comment by faddat** [comment]
I think it is linted now

File: .golangci.yml [hunk 16]
```diff
@@ -4,13 +4,90 @@ run:
...
+    - unused
+    - varcheck
+    #    - varnamelen <- disabled because defies idiom
+    #    - wastedassign
+    - whitespace
+#    - wrapcheck
+#    - wsl
 
 issues:
   exclude-rules:
```
Comment (ValarDragon) [inline .golangci.yml:93]: Can we exclude `client/docs` and have those changes undone?

Comment (faddat) [reply .golangci.yml:93]: yep, sure can

File: app/keepers.go [hunk 17]
```diff
@@ -38,17 +38,17 @@ import (
...
 
-	// Params: Parameters that are always available
+	// Params: Parameters that are always available.
 	"github.com/cosmos/cosmos-sdk/x/params"
 	paramskeeper "github.com/cosmos/cosmos-sdk/x/params/keeper"
 	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	paramproposal "github.com/cosmos/cosmos-sdk/x/params/types/proposal"
 
-	// Slashing:
+	// Slashing:.
```
Comment (ValarDragon) [inline app/keepers.go:51]: lolwat

File: app/modules.go [hunk 18]
```diff
@@ -308,7 +304,8 @@ var modulesOrderInitGenesis = []string{
 	wasm.ModuleName,
 }
 
-// simulationModules returns modules for simulation manager
+// simulationModules returns modules for simulation manager.
+//nolint:unparam
 func simulationModules(app *OsmosisApp, encodingConfig appparams.EncodingConfig, skipGenesisInvariants bool) []module.AppModuleSimulation {
```
Comment (ValarDragon) [inline app/modules.go:309]:
```suggestion
func simulationModules(app *OsmosisApp, encodingConfig appparams.EncodingConfig, _skipGenesisInvariants bool) []module.AppModuleSimulation {
```

File: app/upgrades/v4/prop12.go [hunk 19]
```diff
@@ -31,9 +30,7 @@ func Prop12(ctx sdk.Context, bank bankkeeper.Keeper, distr *distrkeeper.Keeper)
 		total += amount
 	}
 
-	//deduct from the feePool tracker
 	feePool := distr.GetFeePool(ctx)
```
Comment (ValarDragon) [inline app/upgrades/v4/prop12.go:33]:
```suggestion
	// deduct from the feePool tracker
	feePool := distr.GetFeePool(ctx)
```

File: cmd/osmosisd/cmd/balances_from_state_export.go [hunk 20]
```diff
@@ -177,18 +177,22 @@ Example:
...
+			clientCtx.Codec.MustUnmarshalJSON(genState["auth"], &authGenesis)
+
+			// This didn't seem to do anything, so have removed.
+			/*
+				accounts, err := authtypes.UnpackAccounts(authGenesis.Accounts)
+				if err != nil {
+					panic(err)
+				}
+				sanitizedAccounts := authtypes.SanitizeGenesisAccounts(accounts)
+			*/
```
Comment (ValarDragon) [inline cmd/osmosisd/cmd/balances_from_state_export.go:189]: Hrmm, can we undo a lot of changes to this file. I think they're going to conflict with other ongoing work.

I may make a commit for this later

File: osmoutils/cli_helpers.go [hunk 21]
```diff
@@ -20,9 +20,9 @@ const (
 	bitlen = 64
 )
 
-func ParseUint64SliceFromString(s string, seperator string) ([]uint64, error) {
-	var parsedInts []uint64
-	for _, s := range strings.Split(s, seperator) {
+func ParseUint64SliceFromString(s string, separator string) ([]uint64, error) {
+	var parsedInts []uint64 //nolint:prealloc
```
Comment (ValarDragon) [inline osmoutils/cli_helpers.go:24]: I think we should remove the prealloc linter, no need to add extra complexity to lots of our simple code

File: proto/osmosis/gamm/v1beta1/query.proto [hunk 22]
```diff
@@ -106,15 +106,15 @@ message QueryTotalSharesResponse {
...
+  uint64 poolID = 1 [ (gogoproto.moretags) = "yaml:\"pool_id\"" ];
 }
 message QueryPoolAssetsResponse {
   repeated PoolAsset poolAssets = 1 [ (gogoproto.nullable) = false ];
 }
 
 //=============================== SpotPrice
 message QuerySpotPriceRequest {
-  uint64 poolId = 1 [ (gogoproto.moretags) = "yaml:\"pool_id\"" ];
+  uint64 poolID = 1 [ (gogoproto.moretags) = "yaml:\"pool_id\"" ];
```
Comment (ValarDragon) [inline proto/osmosis/gamm/v1beta1/query.proto:117]: Is it straightforward for us to delay the Id -> ID change to another PR? I'm worried its going to cause a breaking change that we're not going to easily catch (And makes a bunch of merge conflicts with other things)

**Review by ValarDragon — COMMENTED** [review]
I think we may end up undoing the linters for import order, periods at end of commas, markdown linters, grouping constants/var invocations, but I think we can do that in a future PR.

(The changes are all good! Its just really annoying to have to revise stuff for those, for little value imo. I often disagree with the import re-orders for logical clarity)

File: app/keepers.go (hunk 17 above)
Comment (faddat) [reply app/keepers.go:51]: in fact, this explains a half dozen things.  

thanks!

**Comment by faddat** [comment]
there are currently a number of problems, but `lolwat` was highly illustrative! (fer realsies lol)(

**Comment by faddat** [comment]
hey this is the first time I have seen this behavior:

this "works on my computer"

File: proto/osmosis/gamm/v1beta1/query.proto (hunk 22 above)
Comment (faddat) [reply proto/osmosis/gamm/v1beta1/query.proto:117]: yep we can do that.  I think id just need to change a setting in revive.

File: osmoutils/cli_helpers.go (hunk 21 above)
Comment (faddat) [reply osmoutils/cli_helpers.go:24]: we are of one mind

**Comment by faddat** [comment]
huh.  magic.  Ok, well it builds.  Taking out nolint....

**Comment by faddat** [comment]
@ValarDragon I'll need to get to the poolID and other naming convention changes tomorrow-- I think it'll be time consuming in the other direction, too.

File: proto/osmosis/gamm/v1beta1/query.proto (hunk 22 above)
Comment (faddat) [reply proto/osmosis/gamm/v1beta1/query.proto:117]: actually, I should rephrase-- it isn't striaghtforward at all, est maybe a day's work but I already think that I need to dial back to this commit:

https://github.com/osmosis-labs/osmosis/pull/1036/commits/041617c503fe9f627b6f32bbb64dede751becd8f

File: cmd/osmosisd/cmd/balances_from_state_export.go (hunk 20 above)
Comment (faddat) [reply cmd/osmosisd/cmd/balances_from_state_export.go:189]: I can do like

git checkout main -- cmd/osmosisd/cmd/balances_from_state_export.go

**Comment by ValarDragon** [comment]
I don't think its worth wasting a day of time on reverting the ID changes. We just need to carefully check all the proto changes, and function signature changes, to be confident that nothing people depend on would break

File: proto/osmosis/superfluid/genesis.proto [hunk 23]
```diff
@@ -16,6 +16,8 @@ message GenesisState {
       [ (gogoproto.nullable) = false ];
   repeated SuperfluidIntermediaryAccount intermediary_accounts = 4
       [ (gogoproto.nullable) = false ];
+  repeated LockIDIntermediaryAccountConnection intemediary_account_connections = 5
+      [ (gogoproto.nullable) = false ];
   repeated LockIdIntermediaryAccountConnection intemediary_account_connections =
       5 [ (gogoproto.nullable) = false ];
```
Comment (ValarDragon) [inline proto/osmosis/superfluid/genesis.proto:22]: This is a bug, should not be duplicated

File: store/node.go [hunk 24]
```diff
@@ -212,13 +212,14 @@ func (node Node) accumulate() (res sdk.Int) {
 	return
 }
 
+//nolint:revive
```
Comment (ValarDragon) [inline store/node.go:215]: what is this linter?

File: .gitignore [hunk 25]
```diff
@@ -198,4 +198,5 @@ $RECYCLE.BIN/
 /artifacts/
 /.vscode/
 /scripts/local/
-/x/incentives/keeper/osmosis_testing/
\ No newline at end of file
+/x/incentives/keeper/osmosis_testing/
+v7
```
Comment (ValarDragon) [inline .gitignore:202]: why are we gitignoring this?

File: app/keepers.go [hunk 26]
```diff
@@ -61,68 +61,65 @@ import (
...
 	lockupkeeper "github.com/osmosis-labs/osmosis/v7/x/lockup/keeper"
 	lockuptypes "github.com/osmosis-labs/osmosis/v7/x/lockup/types"
 
-	// Mint: Our modified version of github.com/cosmos/cosmos-sdk/x/mint
+	// Mint: Our modified version of github.com/cosmos/cosmos-sdk/x/mint.
 	mintkeeper "github.com/osmosis-labs/osmosis/v7/x/mint/keeper"
 	minttypes "github.com/osmosis-labs/osmosis/v7/x/mint/types"
 
-	// Pool incentives:
+	// Pool incentives:.
```
Comment (ValarDragon) [inline app/keepers.go:109]: Did we get to the bottom of why these priods after colons / can we fix them / remove alot of the excess periods?

I think we  should disable whatever linter caused this

Comment (ValarDragon) [reply app/keepers.go:109]: like if we grep for `:.` and just revert those back to `:`

File: app/app.go [hunk 27]
```diff
@@ -278,7 +273,7 @@ func NewOsmosisApp(
 	//
 	// NOTE: this is not required apps that don't use the simulator for fuzz testing
 	// transactions
-	app.sm = module.NewSimulationManager(simulationModules(app, encodingConfig, skipGenesisInvariants)...)
+	app.sm = module.NewSimulationManager(simulationModules(app, encodingConfig)...)
```
Comment (ValarDragon) [inline app/app.go:276]: This was removed deu to it being an unused param in underlying function

File: cmd/osmosisd/cmd/testnetify/state_export_to_testnet.go (hunk 5 above)
Comment (ValarDragon) [reply cmd/osmosisd/cmd/testnetify/state_export_to_testnet.go:47]: lets undo this, and just add in _ = operatorValOper.

A lot of the point of this file is starter code to build off of

File: proto/osmosis/txfees/v1beta1/query.proto [hunk 28]
```diff
@@ -5,7 +5,7 @@ import "gogoproto/gogo.proto";
 import "google/api/annotations.proto";
 import "google/protobuf/duration.proto";
 
-import "osmosis/txfees/v1beta1/feetoken.proto";
```
Comment (ValarDragon) [inline proto/osmosis/txfees/v1beta1/query.proto:8]: Was this done by a linter?

File: x/mint/keeper/keeper.go [hunk 29]
```diff
@@ -52,15 +52,15 @@ func NewKeeper(
 }
 
 // SetInitialSupplyOffsetDuringMigration sets the supply offset based on the balance of the
-// Develope rVesting Module Account.  It should only be called one time during the initial
-// migration to v7
+// Develop rVesting Module Account.  It should only be called one time during the initial
```
Comment (ValarDragon) [inline x/mint/keeper/keeper.go:55]:
```suggestion
// Developr Vesting Module Account.  It should only be called one time during the initial
```

File: store/node.go (hunk 24 above)
Comment (faddat) [reply store/node.go:215]: revive replaced.... golint -- it does a whole lot of stuff:

```
Fast, configurable, extensible, flexible, and beautiful linter for Go. Drop-in replacement of golint. Revive provides a framework for development of custom rules, and lets you define a strict preset for enhancing your development & code review processes.
```

File: .gitignore (hunk 25 above)
Comment (faddat) [reply .gitignore:202]: I am confused by that.

I added v7 beause of the protobufs temp folder there.

But I don't recall adding the other.  :/

File: proto/osmosis/txfees/v1beta1/query.proto (hunk 28 above)
Comment (faddat) [reply proto/osmosis/txfees/v1beta1/query.proto:8]: I don’t know and of course that is a problem.

**Comment by faddat** [comment]
Dev, I think some of the entropy here came from my decision to use Goland to do some code formatting.

so I am going to dial this backwards.

figure we can have it rolling in a week.

do you have any commentary on the linter config?  I am thinking of applying this config to the cosmos sdk, and the rolling it into tinyport, which is a starport that doesn’t spy on you (among other things)

**Comment by ValarDragon** [comment]
Tinyport sounds really cool!

We also routinely run into issues with starport, I just don't use it due to the amount of obsolete code it writes, and boilerplate it doesn't handle. I have a better time copy-pasting existing module.go's

**Comment by faddat** [comment]
closing, and starting from main again with a more limited set of linters.

**Pull request closed without merging**